crates/rbuilder/src/primitives/mod.rs

```diff
@@ -711,10 +713,26 @@ impl TransactionSignedEcRecoveredWithBlobs {
         } else if blob_sidecar.is_some() && tx.inner().blob_versioned_hashes().is_none() {
             Err(TxWithBlobsCreateError::BlobsMissingEip4844)
         // Groovy!
+        // No blob txs at all
+        } else if let Some(b_sidecar) = blob_sidecar {
+            match b_sidecar {
+                BlobTransactionSidecarVariant::Eip4844(sidecar) => Ok(Self {
+                    tx,
+                    blobs_sidecar: Arc::new(BlobTransactionSidecarVariant::Eip4844(sidecar)),
+                    metadata: metadata.unwrap_or_default(),
+                }),
+                BlobTransactionSidecarVariant::Eip7594(sidecar) => Ok(Self {
+                    tx,
+                    blobs_sidecar: Arc::new(BlobTransactionSidecarVariant::Eip7594(sidecar)),
+                    metadata: metadata.unwrap_or_default(),
+                }),
+            }
         } else {
             Ok(Self {
                 tx,
-                blobs_sidecar: Arc::new(blob_sidecar.unwrap_or_default()),
+                blobs_sidecar: Arc::new(BlobTransactionSidecarVariant::Eip4844(
+                    BlobTransactionSidecar::default(),
+                )),
```

Comment:
We ideally need to know which fork we are currently at, in order to create an EIP-4844 type sidecar or EIP-7954 sidecar. We unfortunately don't have this info here.